feat: add initial interfaces [SC-6425]

natrad100 opened the pull request.

natrad100 committed:
feat: add a few interfaces

natrad100 committed:
chore: shift to situational_awareness

natrad100 committed:
chore: update top_level package information

natrad100 committed:
Merge branch 'main' of github.com:Greenroom-Robotics/maritime_interfaces into draft_interfaces

russkel commented on maritime_interfaces/sensors/battery/Battery.msg:
```diff
@@ -0,0 +1,3 @@
+std_msgs/Header header
```
Yeah this doesn't have any value. Should get rid of this message.

russkel commented on maritime_interfaces/sensors/generic/Pressure.msg:
```diff
@@ -0,0 +1,3 @@
+std_msgs/Header header
```
Get rid of all thisc message.

russkel commented on maritime_interfaces/engineering/engine/EngineState.msg:
```diff
@@ -0,0 +1,11 @@
+float32 speed
+float32 throttle
+maritime_interfaces_support/GenericSensorData turbo_state
+maritime_interfaces_support/GenericSensorData intake_state
```
Don't have generic sensor data. This is not ROS2 convention.

russkel commented on maritime_interfaces/support/shared/FluidData.msg:
```diff
@@ -0,0 +1,5 @@
+# FluidData.msg
```
Flatten this message.

russkel commented on maritime_interfaces/sensors/generic/Temperature.msg:
```diff
@@ -0,0 +1,3 @@
+std_msgs/Header header
```
Get rid of this message.

russkel commented on maritime_interfaces/support/shared/GenericSensorData.msg:
```diff
@@ -0,0 +1,4 @@
+std_msgs/Header header
```
Remove this message.

russkel commented on maritime_interfaces/support/shared/ComponentInfo.msg:
```diff
@@ -0,0 +1,9 @@
+# ComponentInfo.msg
```
Does this exist as a CAN message already?

natrad100 replied:
It does, but doesn't mean we don't want something on the other side of the integration service.

natrad100 committed:
docs: added list of maritime interfaces
- removed 'relay' style messages

natrad100 committed:
feat: flattened shared msgs
- removed GenericSensorData
- removed ContainerState
- updated FluidData (flattened)
- updated EngineState

natrad100 committed:
feat: updated package dependencies
- note many of these are private packages at current

natrad100 committed:
chore: add private repo publish

natrad100 committed:
lint

natrad100 committed:
chore: updated names for workflows

natrad100 committed:
chore: misc hyperlink fixes README.md

natrad100 committed:
docs: github linking issues with readme
